x/evm/artela/api/evm_api.go

```diff
@@ -44,17 +42,18 @@ func GetEvmHostInstance(ctx context.Context) (coretypes.EvmHostApi, error) {
 		return nil, errors.New("GetEvmHostInstance: unwrap AspectRuntimeContext failed")
 	}
 	return &evmHostApi{
-		aspectCtx:      aspectCtx,
-		getCtxByHeight: evmHostInstance.getCtxByHeight,
-		ethCall:        evmHostInstance.ethCall,
+		aspectCtx: aspectCtx,
+		ethCall:   evmHostInstance.ethCall,
 	}, nil
 }
 
 func (e evmHostApi) StaticCall(ctx *coretypes.RunnerContext, request *coretypes.EthMessage) *coretypes.EthMessageCallResult {
-	sdkCtx, err := e.getCtxByHeight(ctx.BlockNumber, false)
-	if err != nil {
-		return coretypes.ErrEthMessageCallResult(err)
+
+	aspectCtx, ok := ctx.Ctx.Value(artelatypes.AspectContextKey).(*artelatypes.AspectRuntimeContext)
```

Comment:
The RunnerContext stores AspectRuntimeContext rather than cosmos.Context. Retrieving values directly from the stored object is not appropriate. Please ensure that this logic is correct, and the test cases cover this code.